x-pack/plugins/maps/server/data_indexing/get_indexes_matching_pattern.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IScopedClusterClient } from 'kibana/server';
+import { CatIndicesRecord } from '@elastic/elasticsearch/api/types';
+import { MatchingIndexesResp } from '../../common';
+
+export async function getMatchingIndexes(
+  indexPattern: string,
+  { asCurrentUser }: IScopedClusterClient
+): Promise<MatchingIndexesResp> {
+  try {
+    const { body: indexResults } = await asCurrentUser.cat.indices({
+      index: indexPattern,
+      format: 'JSON',
+    });
+    const matchingIndexes = indexResults
+      .map((indexRecord: CatIndicesRecord) => indexRecord.index)
+      .filter((indexName) => !!indexName);
+    return {
+      success: true,
+      matchingIndexes: matchingIndexes as string[],
+    };
+  } catch (error) {
+    return {
+      success: false,
```

Comment:
MatchingIndexesResp type declares an error key. Either set the error key here or remove the error key from the type